### ISSUE TITLE
Remove escape characters from JSON.GET

### ISSUE THREAD

**Issue by aboimpinto** [tombatron/NReJSON#6]
I have this code 

            var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync("localhost");
            var redisDb =  connectionMultiplexer.GetDatabase(1);

            var setRedisResult = await redisDb
                .JsonSetAsync("test", "{ \"param1\": \"value1\", \"param2\": \"value2\", \"intValue\": 1974 }")
                .ConfigureAwait(false);

            var getRedisResult = await redisDb
                .JsonGetAsync("test", ".param1")
                .ConfigureAwait(false);

            var getResult = getRedisResult.ToString().Replace("\"", ""); 
            if (getResult == "value1")
            {
                Console.WriteLine("YES!!");
            }

            Console.WriteLine("nooooooooooooooooooooooooooooooooo");

What I would like to know is,  why the JsonGetAsync doesn't return the string with the \" in the beginning and end of the key and value?
I have to replace it in my code but this could be done internally and if not why?

I checked the tests and there is no test checking the content of what we get. 

Best wishes
Paulo Aboim Pinto

**Comment by tombatron** [comment]
Hi there Paulo!

I think the crux of the issue is that RedisJson makes no attempt to negotiate the underlying data types of JSON payloads therefore the NReJSON extensions don't either (ie. it doesn't do any kind of parsing or unescaping). 

Take the following series of Redis commands:

![Screenshot from 2020-06-05 18-19-06](https://user-images.githubusercontent.com/119369/83926951-55b3dc00-a759-11ea-884b-d7f6cbc05828.png)

As you can see, NReJSON is only interfacing with Redis and giving back the results of the RedisJson extension commands as it gets them. 

Could it be done internally? Sure! But we're not because the intent here was only to provide a convenient means of interfacing with the plugin.

**Comment by aboimpinto** [comment]
ok... ;)

**Comment by tombatron** [comment]
I'm reopening this because I think it would be helpful to add a way to deal with the raw JSON results. I'm thinking I will provide a series of overloads that will interact with a serializer provided by the user. 

So maybe I would define a generic serializer interface like:

```csharp
public interface ISerializer
{
        TResult Deserialize<TResult>(RedisResult value);
}
```

The consumer of the library could then choose whatever JSON serializer they like (thereby not tying them to my opinions on such things). For example, if you are using `System.Text.Json` you're implementation might look like:

```csharp
public class Serializer : ISerializer
{
    public TResult Deserialize<TResult>(RedisResult value) =>
        JsonSerializer.Deserialize<TResult>(value.ToString());
}
```

Then I would add a static property somewhere where you could define the serializer for your project.

Next I would define the overloads that would interface with that serializer:

```csharp
public static Task<TResult> JsonGetAsync<TResult>(this IDatabase db, RedisKey key, params string[] paths) =>
    db.JsonGetAsync<TResult>(key, false, default, null, null, paths);

public static async Task<TResult> JsonGetAsync<TResult>(this IDatabase db, RedisKey key, bool noEscape = false, string indent = default, string newline = default, string space = default, params string[] paths)
{
    var queryResult = await db.JsonGetAsync(key, noEscape, indent, newline, space, paths).ConfigureAwait(false);

    return Serializer.Deserialize<TResult>(queryResult);
}
```

With all that in place the example you provided would look something like:

```csharp
var result = await _db.JsonGetAsync<string>("test", ".param1");
```
...and `result` would be equal to `value1` instead of `"value1"`.

**Comment by aboimpinto** [comment]
Like the solution.
It doesn't interfere with the current behavior but extends it if the consumer wants.

**Comment by tombatron** [comment]
Yeah my biggest hang up was I didn't want to make assumptions about how people wanted to handle JSON. For example in a lot of projects that I work with in my professional life only use `System.Text.Json`.

**Comment by tombatron** [comment]
@aboimpinto Hey Paulo, I just published version 3.0 (I made some breaking changes so I went up a major version) that should enable you to hook in whatever JSON serializer your project is using (the example I have uses System.Text.Json) to NReJSON. 

If you notice anything wonky let me know!

Thanks,

Tom
